includes/controller/api.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/**
+ * Creates a link or url to a given resource.
+ *
+ * @param $resource
+ */
+function api_link($resource)
+{
+    return page_link_to('api') . '&r=' . $resource;
+}
+
+function getApiShifts()
+{
+    $pageSize = isset($_REQUEST['per_page']) && preg_match('/\d+/', $_REQUEST['per_page']) ? (int)$_REQUEST['per_page'] : null;
+    $page = isset($_REQUEST['page']) && preg_match('/\d+/', $_REQUEST['page']) ? (int)$_REQUEST['page'] : null;
+    $limit = '';
+    if (null !== $page && null !== $pageSize) {
+        $offset = $page === 1 ? 0 : $page === 2 ? $pageSize : ($page - 1)*$pageSize;
+        $limit = sprintf('LIMIT %s, %s', $pageSize, $offset);
+    }
+
+    $query = "SELECT
+                s.*, r.Name as locationName, r.location, r.lat, r.long, t.name as shiftType
+              FROM `Shifts` s
+              JOIN `ShiftTypes` t ON s.`shifttype_id` = t.`id`
+              JOIN `Room` r ON s.`RID` = r.`RID`
+              WHERE s.`start` > UNIX_TIMESTAMP() OR s.`end` > UNIX_TIMESTAMP()
+              GROUP  BY s.`SID`
+              ORDER BY s.`start`
+              " . $limit;
```

Comment:
@ElectricMaxxx The syntax for LIMIT is:
```SQL
 [LIMIT {[offset,] row_count | row_count OFFSET offset}]
```
So you probably want to try ```LIMIT 0,10``` for the first 10 results.